### ISSUE TITLE
compiler crash after updating to 5.0.8

### ISSUE THREAD

**Issue by mscherer82** [trufflesuite/truffle#1813]
## Issue

`truffle compile` causes a memory error. In version `5.0.7` compiling works fine.

## Actual Results

```
λ truffle compile

Compiling your contracts...
===========================

<--- Last few GCs --->

[14948:000001CE41908F60]    77537 ms: Mark-sweep 1693.0 (1760.7) -> 1692.9 (1729.7) MB, 1169.6 / 0.0 ms  last resort GC in old space requested
[14948:000001CE41908F60]    78715 ms: Mark-sweep 1692.9 (1729.7) -> 1692.9 (1729.7) MB, 1178.1 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 000001EBEFB25879 <JSObject>
    1: next [C:\Projects\everdragons\everdragons-contract\node_modules\babylon\lib\index.js:~682] [pc=0000000B3126C3FF](this=000003453A6DECC1 <Parser map = 000002D5FFC88559>)
    2: /* anonymous */ [C:\Projects\everdragons\everdragons-contract\node_modules\babylon\lib\index.js:~3507] [pc=0000000B3128FB00](this=000003453A6DECC1 <Parser map = 000002D5FFC88559>,base=000002E9072D4C79 <Node map = 0...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node_module_register
 2: v8::internal::FatalProcessOutOfMemory
 3: v8::internal::FatalProcessOutOfMemory
 4: v8::internal::Factory::NewUninitializedFixedArray
 5: v8::internal::WasmDebugInfo::SetupForTesting
 6: v8::internal::interpreter::BytecodeArrayRandomIterator::UpdateOffsetFromIndex
 7: 0000000B310843C1
```

## Environment

* Operating System: Windows 10
* Truffle version (`truffle version`): 5.0.8
* node version (`node --version`): 8.12.0

**Comment by ccolorado** [comment]
Was getting ready to submit this test:
Steps to reproduce
I created a blank virtual machine with ubuntu ubuntu-18.04.2-live-server-amd64.iso
install nvm 
nvm install v10.15.0
nvm alias default v10.15.0
npm install -g truffle
npm install -g ganache-cli
clone any solidity truffle project
npm install
truffle test
Get this result

![Screenshot_20190313_163850](https://user-images.githubusercontent.com/876976/54319434-2a226000-45af-11e9-82a2-44d00ca44bc5.png)

**Comment by eggplantzzz** [comment]
Can you give me a specific codebase or Solidity file that fails? I unboxed the Truffle metacoin box and it compiles just fine.

**Comment by hardlydifficult** [comment]
Here's an example from greenkeeper on a public repo https://github.com/unlock-protocol/unlock/issues/2067

**Comment by eggplantzzz** [comment]
Ok, so I did a lot of testing and there is some conflict with `babel-register`. It is somehow causing `require('path/to/cached/compiler)` to hang in the loading strategy [here](https://github.com/trufflesuite/truffle/blob/develop/packages/truffle-compile/compilerSupplier/loadingStrategies/VersionRange.js#L37) and blow the heap. When I comment out `require('babel-register')` in the Truffle config, it is able to successfully compile. I'm not sure what we're going to do about this but we'll try and see what we can do for these to peacefully coexist :)

**Comment by eggplantzzz** [comment]
Ah ok so all this makes a lot more sense now. babel-register is binding to Node's require function and processing all `.js` files that it requires. The Solidity compiler files that Truffle requires are `.js` files...and they are quite big.

We are going to try and find a nice fix for this, but in the meantime as a workaround it looks like babel-register allows an `ignore` option to be uses.

So you could do something like `require('babel-register')({ ignore: /*soljson*/ })` and it should work.

**Comment by eggplantzzz** [comment]
So we provided a fix for this in [version 5.0.10](https://github.com/trufflesuite/truffle/releases/tag/v5.0.10) that was just released. You should be able to get away with using babel-register without specifying the `ignore` option. Try it out and let me know if there are still some problems to address. Thanks!
